integration/mediation-tests/tests-transport/src/test/java/org/wso2/carbon/esb/vfs/transport/test/VFSHidePasswordLogESBJAVA3419.java

```diff
@@ -26,21 +24,54 @@
 
 public class VFSHidePasswordLogESBJAVA3419 extends ESBIntegrationTest {
 
+    private CarbonLogReader logReader;
+    private static String PROXY1_NAME = "HidePasswordListenerProxy";
+    private static String PROXY2_NAME = "HidePasswordSenderProxy";
+
     @BeforeClass(alwaysRun = true)
     public void init() throws Exception {
         super.init();
+        logReader = new CarbonLogReader();
+        logReader.start();
+        deployArtifacts();
     }
 
     @AfterClass(alwaysRun = true)
     public void restoreServerConfiguration() throws Exception {
-        super.cleanup();
+        logReader.stop();
+        Utils.undeploySynapseConfiguration(PROXY1_NAME, Utils.ArtifactType.PROXY, false);
+        Utils.undeploySynapseConfiguration(PROXY2_NAME, Utils.ArtifactType.PROXY, true);
+
     }
 
     @SetEnvironment(executionEnvironments = { ExecutionEnvironment.STANDALONE })
     @Test(groups = { "wso2.esb" }, description = "Checking VFSTransportListener not logs the clear password on error")
     public void testVFSListenerHidePasswordInLog() throws Exception {
 
-        addProxyService(AXIOMUtil.stringToOM(
+        Assert.assertFalse(logReader.checkForLog("ClearPassword", 5),
+                " The password is getting printed in the log in the VFSTransportListener.");
+        logReader.clearLogs();
+    }
+
+    @SetEnvironment(executionEnvironments = { ExecutionEnvironment.STANDALONE })
+    @Test(groups = { "wso2.esb" }, description = "Checking VFSTransportSender not logs the clear password on error",
+            dependsOnMethods = "testVFSListenerHidePasswordInLog")
+    public void testVFSSenderHidePasswordInLog() throws Exception {
+        try {
+            OMElement response = axis2Client
+                    .sendSimpleStockQuoteRequest(getProxyServiceURLHttp("HidePasswordSenderProxy"),
+                            getBackEndServiceUrl(ESBTestConstant.SIMPLE_STOCK_QUOTE_SERVICE), "WSO2");
+        } catch (AxisFault e) {
```

Comment:
Shouldn't we avoid catching the exception since this is a test case?

Reply:
This test case expects failure, this test a failure scenario

Reply:
:+1: Maybe JUnit test rules is something we can use here. 
1. https://www.infoq.com/news/2009/07/junit-4.7-rules